dev/NavigationView/NavigationView.cpp

```diff
@@ -154,6 +152,58 @@ NavigationView::NavigationView()
 
     Unloaded({ this, &NavigationView::OnUnloaded });
     Loaded({ this, &NavigationView::OnLoaded });
+
+    m_selectionModel.SingleSelect(true);
+    m_selectionChangedEventToken = m_selectionModel.SelectionChanged(winrt::auto_revoke, { this, &NavigationView::OnSelectionModelSelectionChanged });
+
+    m_navigationViewItemsFactory = winrt::make_self<NavigationViewItemsFactory>();
+}
+
+void NavigationView::OnSelectionModelSelectionChanged(winrt::SelectionModel selectionModel, winrt::SelectionModelSelectionChangedEventArgs e)
+{
+    auto selectedItem = selectionModel.SelectedItem();
+    auto selectedIndex = selectionModel.SelectedIndex();
+
+    // Ignore this callback if the SelectedItem property of NavigationView is already set to the item
+    // being passed in this callback. This is because the item has already been selected
+    // via API and we are just updating the m_selectionModel state to accurately reflect the new selection.
+    // TODO: Update SelectedItem comparison to work for the exact same item datasource scenario
+    if (m_shouldIgnoreNextSelectionChange ||
```

Comment:
We should file a bug on SelectionModel if it is raising selection changed event when the selection has not changed.

Reply:
The motivation for this comparison is not due to SelectionModel raising the event when the selection has not changed, but rather correctly raising the event, however only because we internally updated it to reflect the selectionstate of what was set via NavigationView API